File: Fireworks.UI/RandomFloat.cs
using System.Numerics;

namespace Fireworks.UI;

/// <summary>
/// Helper class providing methods for generating random floating-point numbers.
/// </summary>
static class RandomFloat
{
    public static float NextFloat(float min, float max) =>
        Random.Shared.NextSingle() * (max - min) + min;

    public static Vector2 NextVector()
    {
        var angle = NextFloat(0, MathF.PI * 2);
        return new Vector2(MathF.Cos(angle), MathF.Sin(angle));
    }
}

File: Fireworks.Web/randomFloat.js
/**
 * Helper class providing methods for generating random floating-point numbers.
 */
class RandomFloat {
    /**
     * Generate a random float between min and max
     * @param {number} min - Minimum value
     * @param {number} max - Maximum value
     * @returns {number} Random float between min and max
     */
    static nextFloat(min, max) {
        return Math.random() * (max - min) + min;
    }

    /**
     * Generate a random unit vector (direction)
     * @returns {Object} Vector with x and y components
     */
    static nextVector() {
        const angle = RandomFloat.nextFloat(0, Math.PI * 2);
        return {
            x: Math.cos(angle),
            y: Math.sin(angle)
        };
    }

    /**
     * Generate a random integer between min and max (inclusive)
     * @param {number} min - Minimum value
     * @param {number} max - Maximum value
     * @returns {number} Random integer between min and max
     */
    static nextInt(min, max) {
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }
}
